README.md

```diff
@@ -113,25 +113,28 @@ Creating artifacts/abis/ to store compilation artifacts
 
 ### `deploy`
 
-> NOTICE: this method doesn't use an account, which will be deprecated very soon as StarkNet makes deployments from accounts mandatory.
+> NOTICE: Calling this method will require an account very soon (using the `--account` param) as StarkNet will make deployments from accounts mandatory.
 
 > Token for deployments to Alpha Mainnet can be set with the `--token` option.
 
 ```sh
-nile deploy contract --alias my_contract
+nile deploy contract --account setup_account_alias --alias my_contract
```

Comment:
although i like the `--account` flag much better, for consistency with other commands we should use pk aliases
```sh
nile deploy PK_ALIAS contract ...
```
#212 partially addresses it. having an `--account` flag is also the way because it would maintain the API even after having default accounts with https://github.com/OpenZeppelin/nile/issues/73

Reply:
i understand this makes much more difficult to identify therefore support both deployments (with/without account), unclear what's best here

Reply:
My first thought was the same (to follow the same format of send), but as you state, the difference here is that we need to support both with and without account options. Click parameters (instead of options) are not optional (they could be, but with important restrictions I think are not worth to address). 

IMO the best approach would be to let account an option in this PR (because it is potentially the desired behavior anyway) and fix (or improve) pk handling later in the issues you mentioned.

Reply:
> the difference here is that we need to support both
i'm not sure about this. "accountless" deployments are already discouraged and will be prohibited soon. UDC-like deployments are already functional, so we don't really need to support it.

i think it's preferable to have a consistent, expectable API than an inconsistent one with perfect/final portions. i think we should aim for consistency and eventually migrate all methods to use `--account`.

Reply:
Are you suggesting then removing the support for deploys without accounts? If we remove this support, the current starknet-devnet version we support in Nile (pre 0.10.1) won't work for deployments because it has not predeployed UDC.

Reply:
yes that's the suggestion, since this release will require devnet@0.3.5 already. happy to keep the functionality as long as it doesn't compromise API consistency if there's such alternative

Reply:
I changed the API to match the `send` one, but let a `--ignore_account` flag to deploy without an account with the deprecation warning until the direct deploys are fully removed.

Reply:
thanks. i asked the starkware team, they expect to remove it by this week.